Index: openjdk/jdk/make/java/nio/Makefile
===================================================================
--- openjdk/jdk/make/java/nio/Makefile	2013-10-01 21:20:11.000000000 -0700
+++ openjdk/jdk/make/java/nio/Makefile	2013-10-01 21:30:40.901837422 -0700
@@ -918,7 +918,7 @@
 ifeq ($(PLATFORM), macosx)
   NIO_CC=$(HOST_CC)
 else
-  NIO_CC=$(CC)
+  NIO_CC=$(CC_FOR_BUILD)
 endif
 
 $(GENSOR_EXE) : $(TEMPDIR)/$(GENSOR_SRC)
@@ -934,12 +934,8 @@
 $(SCH_GEN)/SocketOptionRegistry.java: $(GENSOR_EXE)
 	$(prep-target)
 	NAWK="$(NAWK)" SH="$(SH)" $(SH) -e addNotices.sh "$(SOR_COPYRIGHT_YEARS)" > $@
-ifdef CROSS_COMPILE_ARCH
-	$(QEMU) $(GENSOR_EXE) >> $@
-else
 	$(GENSOR_EXE) >> $@
 endif
-endif
 #
 # Generated sun.nio.cs SingleByte classes
 #
@@ -973,12 +969,8 @@
 $(SFS_GEN)/UnixConstants.java: $(GENUC_EXE)
 	$(prep-target)
 	NAWK="$(NAWK)" SH="$(SH)" $(SH) -e addNotices.sh "$(GENUC_COPYRIGHT_YEARS)" > $@
-ifdef CROSS_COMPILE_ARCH
-	$(QEMU) $(GENUC_EXE) >> $@
-else
 	$(GENUC_EXE) >> $@
 endif
-endif
 
 GENSC_SRC = $(PLATFORM_SRC)/native/sun/nio/fs/genSolarisConstants.c
 
@@ -999,11 +991,7 @@
 $(SFS_GEN)/SolarisConstants.java: $(GENSC_EXE)
 	$(prep-target)
 	NAWK="$(NAWK)" SH="$(SH)" $(SH) -e addNotices.sh "$(GENSC_COPYRIGHT_YEARS)" > $@
-ifdef CROSS_COMPILE_ARCH
-	$(QEMU) $(GENSC_EXE) >> $@
-else
 	$(GENSC_EXE) >> $@
 endif
-endif
 
 .PHONY: sources 
